### PR TITLE
[ENG-982] Added type validation for operators

### PR DESCRIPTION
**Pull request by JadeCara**
Closes [ENG-982]

### Description Of Changes

The goal of this work is to enhance the conditional dependencies system by adding type validation for operators and introducing new string-based operators for more flexible condition evaluation.
**Benefits**
Type Safety: Prevents runtime errors from invalid operator usage
Flexibility: New string operators enable more sophisticated text-based conditions
Maintainability: Clear data type compatibility matrix for future operator additions
Reliability: Better error handling and validation throughout the evaluation process

**Key Changes**

1. Enhanced Type Safety for Numeric Operators
Added numbers import for proper type checking
Updated numeric comparison operators (lt, lte, gt, gte) to validate that operands are numeric types before comparison
Prevents type errors when comparing non-numeric values with numeric operators
2. New String Operators
Added three new string-based operators to the Operator enum:
starts_with: Checks if a string starts with a specified prefix
ends_with: Checks if a string ends with a specified suffix
contains: Checks if a string contains a specified substring
3. Data Type Compatibility Matrix
Introduced data_type_operator_compatibility dictionary that maps data types to their compatible operators:
Numeric types (integer, float, double, long): Support equality and comparison operators
Boolean: Supports equality and existence operators
String/Text: Supports equality, string operations, and existence operators
Array: Supports equality, list operations, and existence operators
Object: Supports equality and existence operators
4. Improved Error Handling
Added ConditionEvaluationError exception class for better error reporting
Updated _apply_operator method to raise exceptions instead of returning False for unknown operators
Added try-catch blocks to handle TypeError and ValueError during operator evaluation
5. Enhanced List Operations
Improved not_in_list operator to handle both single values and lists as the first operand
Better handling of edge cases for list operations
Testing
Added comprehensive test coverage for new string operators
Updated existing tests to reflect new error handling behavior
Added integration tests with real-world data scenarios
Added edge case testing for empty strings, None values, and type mismatches

### Code Changes

* src/fides/api/task/conditional_dependencies/evaluator.py: Core evaluation logic and type validation
* src/fides/api/task/conditional_dependencies/schemas.py: Added new string operators to enum
* tests/api/task/conditional_dependencies/test_conditional_dependencies_evaluator.py: Comprehensive test coverage

### Steps to Confirm

1.  This is not hooked up to the front end yet. I have manually tested with https://github.com/ethyca/fidesplus/pull/2408 to verify that checks are working when conditions are being created. 

### Pre-Merge Checklist

* [ ] Issue requirements met
* [ ] All CI pipelines succeeded
* [ ] `CHANGELOG.md` updated
  * [ ] Add a https://github.com/ethyca/fides/labels/db-migration label to the entry if your change includes a DB migration
  * [ ] Add a https://github.com/ethyca/fides/labels/high-risk label to the entry if your change includes a high-risk change (i.e. potential for performance impact or unexpected regression) that should be flagged
  * [ ] Updates unreleased work already in Changelog, no new entry necessary
* Followup issues:
  * [ ] Followup issues created
  * [ ] No followup issues
* Database migrations:
  * [ ] Ensure that your downrev is up to date with the latest revision on `main`
  * [ ] Ensure that your `downgrade()` migration is correct and works
    * [ ] If a downgrade migration is not possible for this change, please call this out in the PR description!
  * [ ] No migrations
* Documentation:
  * [ ] Documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] Documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
  * [ ] If there are any new client scopes created as part of the pull request, remember to update public-facing documentation that references our scope registry
  * [ ] No documentation updates required

[ENG-982]: https://ethyca.atlassian.net/browse/ENG-982?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/fides/api/task/conditional_dependencies/evaluator.py [hunk 1]
```diff
@@ -13,19 +14,75 @@
...
+
+# Numeric comparison operators
+NUMERIC_OPERATORS = {Operator.lt, Operator.lte, Operator.gt, Operator.gte}
+
+# String-specific operators
+STRING_OPERATORS = {
+    Operator.contains,
+    Operator.starts_with,
+    Operator.ends_with,
+    Operator.list_contains,
```
Comment (galvana) [inline src/fides/api/task/conditional_dependencies/evaluator.py]: Is list contains strictly a string operator? I imagine we would want to check if a numeric value is present in a numeric array

File: src/fides/api/task/conditional_dependencies/evaluator.py [hunk 2]
```diff
@@ -13,19 +14,75 @@
...
+    ),
+    Operator.gt: lambda a, b: (
+        a > b if a is not None and isinstance(a, numbers.Number) else False
+    ),
+    Operator.gte: lambda a, b: (
+        a >= b if a is not None and isinstance(a, numbers.Number) else False
+    ),
     Operator.list_contains: lambda a, b: b in a if isinstance(a, list) else False,
-    Operator.not_in_list: lambda a, b: a not in b if isinstance(b, list) else True,
+    Operator.not_in_list: lambda a, b: b not in a if isinstance(a, list) else False,
```
Comment (galvana) [inline src/fides/api/task/conditional_dependencies/evaluator.py]: Is it possible for both the `a` and `b` params to `Operator.not_in_list` to be a list? Meaning you want to check if all the values in one list are present in another?

File: src/fides/api/task/conditional_dependencies/evaluator.py [hunk 3]
```diff
@@ -13,18 +13,9 @@
     Operator,
 )
```
Comment (JadeCara) [inline src/fides/api/task/conditional_dependencies/evaluator.py:15]: The operators were getting pretty meaty. I decided to move them to their own file to reduce  cognitive overhead when considering these files. 

I also created a new test file for operators.

File: src/fides/api/task/conditional_dependencies/operators.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        set(a).issuperset(set(b))
+        if isinstance(a, list) and isinstance(b, list)
+        else False
+    ),
+    Operator.list_disjoint: lambda a, b: (
+        # Check if lists have no common elements
+        set(a).isdisjoint(set(b))
+        if isinstance(a, list) and isinstance(b, list)
+        else False
+    ),
```
Comment (JadeCara) [inline src/fides/api/task/conditional_dependencies/operators.py]: I added new list operators to cover the various use cases I could think of more explicity.

File: src/fides/api/task/conditional_dependencies/operators.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    Operator.list_contains,  # Element in list
+    Operator.not_in_list,  # Element not in list
+    Operator.list_intersects,  # Any common elements
+    Operator.list_subset,  # Column list ⊆ user list
+    Operator.list_superset,  # Column list ⊇ user list
+    Operator.list_disjoint,  # No common elements
+}
+
+# Define data type compatibility with operators
+data_type_operator_compatibility = {
```
Comment (galvana) [inline src/fides/api/task/conditional_dependencies/operators.py:135]: We're not using this anywhere at the moment. Is this for later or included here as a reference?

File: tests/api/task/conditional_dependencies/test_operators.py [hunk 6]
```diff
@@ -0,0 +1,555 @@
...
+                id="not_in_list_anything",
+            ),
+            param(
+                "not_a_list",
+                Operator.not_in_list,
+                "test",
+                False,
+                id="not_in_list_test_no_list",
+            ),
+        ],
```
Comment (galvana) [inline tests/api/task/conditional_dependencies/test_operators.py:463]: I think I found an issue, this missing scenario doesn't pass. Let me know what you think
```
param(
    "admin",
    Operator.not_in_list,
    [],
    True,
    id="not_in_list_empty_user_list_scalar",
)
```

File: src/fides/api/task/conditional_dependencies/operators.py (hunk 5 above)
Comment (JadeCara) [reply src/fides/api/task/conditional_dependencies/operators.py:135]: This is being used in the fidesplus when it verifies the field data type and value before creating the dependency. I can also see it being used for the policy conditionals we have coming up soon too.

File: tests/api/task/conditional_dependencies/test_operators.py (hunk 6 above)
Comment (JadeCara) [reply tests/api/task/conditional_dependencies/test_operators.py:463]: Added this and few other empty list/None scenarios.

File: src/fides/api/task/conditional_dependencies/operators.py [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+#   * None in [] returns False
+#   * None not in [] returns True
+#   * This allows None to be a valid list element for membership testing
+operator_methods = {
+    # Basic operators - handle None naturally using Python's built-in behavior
+    Operator.exists: lambda a, _: a is not None,
+    Operator.not_exists: lambda a, _: a is None,
+    Operator.eq: py_operator.eq,
+    # None == None returns True, None == "anything" returns False
+    Operator.neq: py_operator.ne,
```
Comment (JadeCara) [inline src/fides/api/task/conditional_dependencies/operators.py]:
```suggestion
```
